safekit-cluster-farm/nestedtemplates/cluster.json

```diff
@@ -0,0 +1,653 @@
+		"ips": [
+			"10.0.0.10",
+			"10.0.0.11",
+			"10.0.0.12",
+			"10.0.0.13"
+		],
+		"sktemplate": "[concat(uri(parameters('_artifactsLocation'), 'nestedtemplates/safekitdepl.json'),parameters('_artifactsLocationSasToken'))]",
+		"skclustertemplate": "[concat(uri(parameters('_artifactsLocation'), 'nestedtemplates/installCluster.json'),parameters('_artifactsLocationSasToken'))]",
+		"createviptemplate": "[concat(uri(parameters('_artifactsLocation'),'nestedtemplates/createvip.json'),parameters('_artifactsLocationSasToken'))]",
+		"skmoduletemplate": "[concat(uri(parameters('_artifactsLocation'),'nestedtemplates/installModule.json'),parameters('_artifactsLocationSasToken'))]",
+		"NSGName": "SafeKit-NSG",
+		"storageAccountName": "[concat(uniquestring(resourceGroup().id,parameters('location')), 'stor')]",
+		"lbPoolID": "[if(equals(parameters('LoadBalancer'),'none'),'',resourceId('Microsoft.Network/loadBalancers/backendAddressPools',variables('lbName'),variables('lbPoolName')))]",
+		"lbaddrpool1": [
+			{
+				"id": "[variables('lbPoolID')]"
+			}
+		],
+		"lbaddrpool0": [
+			
+		],
+		"lbaddrpool": "[if(greater(length(variables('lbPoolID')),0),variables('lbaddrpool1'),variables('lbaddrpool0'))]",
+		"WindowsStorageProfile": {
+			"imageReference": {
+				"publisher": "MicrosoftWindowsServer",
+				"offer": "WindowsServer",
+				"sku": "[parameters('OS')]",
+				"version": "latest"
+			},
+			"osDisk": {
+				"createOption": "FromImage"
+			}
+		},
+		"CentosStorageProfile": {
+			"imageReference": {
+				"publisher": "OpenLogic",
+				"offer": "CentOS",
+				"sku": "7.5",
+				"version": "latest"
+			},
+			"osDisk": {
+				"createOption": "fromImage"
+			},
+			"dataDisks": [
+				
+			]
+		},
+		"linuxConfiguration": {
+			"disablePasswordAuthentication": "false"
+		},
+		"addressPrefix": "10.0.0.0/16",
+		"virtualNetworkName": "[concat(resourceGroup().name,'VNET')]",
+		"subnetName": "Subnet",
+		"subnetPrefix": "10.0.0.0/24"
+	},
+	"resources": [
+		{
+			"type": "Microsoft.Network/virtualNetworks",
+			"apiVersion": "2018-08-01",
+			"name": "[variables('virtualNetworkName')]",
+			"location": "[parameters('location')]",
+			"properties": {
+				"addressSpace": {
+					"addressPrefixes": [
+						"[variables('addressPrefix')]"
+					]
+				},
+				"subnets": [
+					{
+						"name": "[variables('subnetName')]",
+						"properties": {
+							"addressPrefix": "[variables('subnetPrefix')]"
+						}
+					}
+				]
+			}
+		},
+		{
+			"type": "Microsoft.Resources/deployments",
+			"apiVersion": "2018-05-01",
+			"name": "optionalVip",
+			"properties": {
+				"mode": "Incremental",
+				"templateLink": {
+					"uri": "[variables('createviptemplate')]",
+					"contentVersion": "1.0.0.0"
+				},
+				"parameters": {
+					"location": {
+						"value": "[parameters('location')]"
+					},
+					"VIPDnsLabel": {
+						"value": "[parameters('VIPDnsLabel')]"
+					}
+				}
+			}
+		},
+		{
+			"type": "Microsoft.Resources/deployments",
```

Comment:
why is this nested?

Reply:
The original implementation allowed publicip & loadbalancer to be optional. Since other resources depends on it I use the deployment resource itself (which always exists even if the loadbalancer/public ip are not to be created) to express the dependency.

Reply:
dependsOn is handled by ARM when using the condition property, so the extra deployment isn't needed... lets remove that.